app/single_resource/views.py

```diff
@@ -45,24 +59,108 @@ def create():
     form = SingleResourceForm()
     if form.validate_on_submit():
-        resource.name = form.name.data
-        resource.address = form.address.data
-        resource.latitude = form.latitude.data
-        resource.longitude = form.longitude.data
+        # field id is not needed for the form, hence omitted with [1:]
+        for field_name in resource_field_names[1:]:
+            setattr(resource, field_name, form[field_name].data)
+        save_associations(resource=resource,
+                          form=form,
+                          descriptors=descriptors,
+                          resource_existed=True)
         try:
             db.session.commit()
             flash('Resource updated', 'form-success')
-            return redirect(url_for('single_resource.edit',
-                                    resource_id=resource_id))
+            return redirect(url_for('single_resource.index'))
         except IntegrityError:
             db.session.rollback()
             flash('Error: failed to save resource. Please try again.',
                   'form-error')
-    for field_name in field_names[1:]:
+    # field id is not needed for the form, hence omitted with [1:]
+    for field_name in resource_field_names[1:]:
         form[field_name].data = resource.__dict__[field_name]
     return render_template('single_resource/edit.html',
                            form=form,
                            resource_id=resource_id)
+
+
+def save_associations(resource, form, descriptors, resource_existed=True):
+    """Save associations from the forms received by 'create' and 'edit' route
+    handlers to the database."""
+    for descriptor in descriptors:
+        if (descriptor.values):
+            AssociationClass = OptionAssociation
+            value = int(form[descriptor.name].data)
+            keyword = 'option'
+        else:
+            AssociationClass = TextAssociation
+            value = form[descriptor.name].data
+            keyword = 'text'
+        association = None
+        if (resource_existed):
+            association = AssociationClass.query.filter_by(
+                resource_id=resource.id,
+                descriptor_id=descriptor.id
+            ).first()
+        if (association is not None):
+            setattr(association, keyword, value)
+        else:
+            arguments = {'resource_id': resource.id,
+                         'descriptor_id': descriptor.id,
+                         keyword: value,
+                         'resource': resource,
+                         'descriptor': descriptor}
+            new_association = AssociationClass(**arguments)
+            db.session.add(new_association)
+
+
+@single_resource.route('/<int:resource_id>/delete', methods=['POST'])
+@login_required
+def delete(resource_id):
+    """Delete a resource."""
+    resource = Resource.query.get(resource_id)
+    if resource is None:
+        abort(404)
+    for descriptor in resource.option_descriptors:
```

Comment:
We shouldn't need this since `cascade='save-update, merge, delete, delete-orphan'` is already set for the `option_descriptors` `relationship`? In other words, when a resource is deleted, so it its descriptors.